packages/actor-query-operation-path-zero-or-more/lib/ActorQueryOperationPathZeroOrMore.ts

```diff
@@ -16,14 +19,59 @@ export class ActorQueryOperationPathZeroOrMore extends ActorAbstractPath {
   }
 
   public async runOperation(path: Algebra.Path, context: ActionContext): Promise<IActorQueryOperationOutputBindings> {
+    context = this.isPathArbitraryLengthDistinct(context);
+    if (context.get('isPathArbitraryLengthDistinct')) {
```

Comment:
Also refer to the static field here.

But now that I've seen this pattern multiple times already, I would suggest abstracting this a bit more.
Instead of having `isPathArbitraryLengthDistinct`, I'd create something like `getDistinctOperation`, which will return a wrapped distinct operation if needed, or undefined if not needed.
Every time you call `getDistinctOperation`, you'd simply have to do an if-check, and return directly if the result is not undefined.

Reply:
The false set on the context will then again have to be out of this abstracted function.

Reply:
Ah right, the context. Maybe you can include it in the return value of the abstracted function instead?

So basically you'd return `{ context, operation }`, where `operation` could be undefined.